### PR TITLE
issue #7410 fixed

### PR DESCRIPTION
**Pull request by shivendra-webkul**
## Issue Reference
#7410

### CODE REVIEW

**Comment by Sanjaybhattwebkul** [comment]
@shivendra-webkul 
This is **working fine.**

**Review by Sanjaybhattwebkul — DISMISSED** [review]

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
It is showing an exception. [Here](https://prnt.sc/qIK-_KKWXwBn)
While related product added product is deleted and the actual product is resaving.

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
It is working fine.

**Comment by devansh-webkul** [comment]
@Sanjaybhattwebkul and @Amitk-webkul,

As per this PR, we have moved all the logic to the controller and that file is I guess in three places. Have you verified all three places?

File: packages/Webkul/Velocity/src/Resources/views/shop/products/view.blade.php [hunk 1]
```diff
@@ -199,6 +199,8 @@
             @include('shop::products.view.related-products')
             
             @include('shop::products.view.up-sells')
+
+            @include('shop::products.view.cross-sells')
```
Comment (devansh-webkul) [inline packages/Webkul/Velocity/src/Resources/views/shop/products/view.blade.php:203]: Remove cross-sell products.

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
It is showing an exception while viewing the shopping cart. [Here](https://prnt.sc/ekUzdbEYTyjq)

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
with the elastic search while viewing the shopping cart it is showing an exception. [Here](https://prnt.sc/Wd-J6GOgKcmY)

**Comment by Amitk-webkul** [comment]
@shivendra-webkul
If a removed product from cross-sell bar in the admin. and open the cart it is showing an exception. [Here](https://prnt.sc/oSXFSF6zlG_1)

**Comment by shivendra-webkul** [comment]
@Amitk-webkul  I checked out this issue but it is not appearing please check video.

Uploading issue 7410.mp4…

**Comment by shivendra-webkul** [comment]
@Amitk-webkul 

https://user-images.githubusercontent.com/121868701/231797616-9b31fdad-8406-4dfd-8bf0-55e352121103.mp4

 I checked out this issue but it is not appearing please check video.

**Comment by shivendra-webkul** [comment]
@Amitk-webkul I also checked out this issue in Elastic Search there is no exception please take reference from this video.

https://user-images.githubusercontent.com/121868701/231944028-2a5c238a-be29-434d-87b6-c94a71ed0222.mp4

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
Please check If cross-sell configuration is [blank](https://prnt.sc/sUt-FG65FwwT). and the product is added to the cart while viewing the cart it is showing an exception. [Here](https://prnt.sc/1wLsXbNxL9ia)

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
If the configuration is blanked it is not showing any product in cross-sell tab. [Here](https://prnt.sc/7VNibE1w5J6M)

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
If the configuration is 0 still it is showing the added product in the cart.

**Comment by Amitk-webkul** [comment]
@shivendra-webkul 
Now it is working fine. [Here](https://prnt.sc/dKSmBCYR88KS)

**Review by Amitk-webkul — APPROVED** [review]

**Review by devansh-webkul — APPROVED** [review]

**Review by jitendra-webkul — APPROVED** [review]

**Pull request merged**